vector/src/main/java/im/vector/riotx/features/settings/devices/DeviceVerificationInfoEpoxyController.kt

```diff
@@ -37,109 +40,324 @@ class DeviceVerificationInfoEpoxyController @Inject constructor(private val stri
+            listener {
+                callback?.onAction(DevicesAction.VerifyMyDevice(cryptoDeviceInfo.deviceId))
+            }
+        }
+    }
 
-            if (info?.deviceId != session.sessionParams.credentials.deviceId) {
-                // Add the delete option
-                dividerItem {
-                    id("d2")
-                }
-                bottomSheetVerificationActionItem {
-                    id("delete")
-                    title(stringProvider.getString(R.string.settings_active_sessions_signout_device))
-                    titleColor(colorProvider.getColor(R.color.riotx_destructive_accent))
-                    iconRes(R.drawable.ic_arrow_right)
-                    iconColor(colorProvider.getColor(R.color.riotx_destructive_accent))
-                    listener {
-                        callback?.onAction(DevicesAction.Delete(info?.deviceId ?: ""))
-                    }
+    private fun addGenericDeviceManageActions(data: DeviceVerificationInfoBottomSheetViewState, cryptoDeviceInfo: CryptoDeviceInfo) {
+        // Offer delete session if not me
+        if (!data.isMine) {
+            // Add the delete option
+            dividerItem {
+                id("d2")
+            }
+            bottomSheetVerificationActionItem {
+                id("delete")
+                title(stringProvider.getString(R.string.settings_active_sessions_signout_device))
+                titleColor(colorProvider.getColor(R.color.riotx_destructive_accent))
+                iconRes(R.drawable.ic_arrow_right)
+                iconColor(colorProvider.getColor(R.color.riotx_destructive_accent))
+                listener {
+                    callback?.onAction(DevicesAction.Delete(cryptoDeviceInfo.deviceId))
                 }
             }
-        } else {
-            loadingItem {
-                id("loading")
+        }
+
+        // Always offer rename
+        dividerItem {
+            id("d3")
+        }
+        bottomSheetVerificationActionItem {
+            id("rename")
+            title(stringProvider.getString(R.string.rename))
+            titleColor(colorProvider.getColorFromAttribute(R.attr.riotx_text_primary))
+            iconRes(R.drawable.ic_arrow_right)
+            iconColor(colorProvider.getColorFromAttribute(R.attr.riotx_text_primary))
+            listener {
+                callback?.onAction(DevicesAction.PromptRename(cryptoDeviceInfo.deviceId))
+            }
+        }
+    }
+
+    private fun handleNonE2EDevice(data: DeviceVerificationInfoBottomSheetViewState) {
+        val info = data.deviceInfo.invoke()
+        genericItem {
+            id("info${info?.deviceId}")
+            title(info?.displayName ?: "")
+            description("(${info?.deviceId})")
+        }
+
+        genericFooterItem {
+            id("infoCrypto${info?.deviceId}")
+            text(stringProvider.getString(R.string.settings_failed_to_get_crypto_device_info))
+        }
+
+        if (!data.isMine) {
```

Comment:
why not calling addGenericDeviceManageActions() ?

Reply:
fixed, thx